File: src/tests/py-simple/counter/simple_counter.py
# Python script to log PID, count, and time to /script-data/container.log every 20 seconds
import os
import time
from datetime import datetime

log_path = "/script-data/container.log"
counter = 0

while True:
    pid = os.getpid()
    now = datetime.now().strftime("%a %b %d %H:%M:%S %Y")
    with open(log_path, "a") as f:
        f.write(f"PID: {pid}, Count: {counter}, Time: {now}\n")
    counter += 1
    time.sleep(20)

File: src/examples/cpp/simple_counter.cpp
// Simple C++ counter that logs to stdout and /script-data/container.log
#include <chrono>
#include <ctime>
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <sys/types.h>
#include <thread>
#include <unistd.h>

int main() {
  unsigned long counter = 0;
  pid_t pid = getpid();
  while (true) {
    auto now = std::chrono::system_clock::now();
    std::time_t t = std::chrono::system_clock::to_time_t(now);
    std::stringstream ss;
    ss << "Counter: " << counter << ", Time: " << std::ctime(&t);
    std::string out = ss.str();
    // ctime adds a newline already
    std::cout << "PID: " << pid << " - " << out << std::flush;
    // also append to /script-data/container.log if available
    try {
      std::ofstream f("/script-data/container.log", std::ios::app);
      if (f) {
        f << "PID: " << pid << " - " << out;
      }
    } catch (...) {
      // ignore
    }
    ++counter;
    std::this_thread::sleep_for(std::chrono::seconds(5));
  }
  return 0;
}
